test_autoarray/dataset/imaging/test_imaging.py

```diff
@@ -0,0 +1,249 @@
+    psf = aa.Kernel2D.ones(shape_native=(3, 3), pixel_scales=1.0)
+
+    imaging = aa.Imaging(
+        image=image, noise_map=noise_map, psf=psf, pad_for_convolver=False
+    )
+
+    assert imaging.image.shape_native == (3, 3)
+    assert imaging.noise_map.shape_native == (3, 3)
+
+    imaging = aa.Imaging(
+        image=image, noise_map=noise_map, psf=psf, pad_for_convolver=True
+    )
+
+    assert imaging.image.shape_native == (5, 5)
+    assert imaging.noise_map.shape_native == (5, 5)
+    assert imaging.image.mask[0, 0] == True
+    assert imaging.image.mask[1, 1] == False
+
+
+def test__noise_covariance_input__noise_map_uses_diag():
+
+    image = aa.Array2D.ones(shape_native=(3, 3), pixel_scales=1.0)
+    noise_covariance_matrix = np.ones(shape=(9, 9))
+
+    imaging = aa.Imaging(image=image, noise_covariance_matrix=noise_covariance_matrix)
+
+    noise_map = aa.Array2D.ones(shape_native=(3, 3), pixel_scales=1.0)
+
+    assert (imaging.noise_map.native == noise_map.native).all()
+
+
+def test__no_noise_map__raises_exception():
+
+    image = aa.Array2D.ones(shape_native=(3, 3), pixel_scales=1.0)
+
+    with pytest.raises(aa.exc.DatasetException):
+
+        aa.Imaging(image=image)
+
+
+def test__from_fits():
+
+    imaging = aa.Imaging.from_fits(
+        pixel_scales=0.1,
+        image_path=path.join(test_data_dir, "3x3_ones.fits"),
+        psf_path=path.join(test_data_dir, "3x3_twos.fits"),
+        noise_map_path=path.join(test_data_dir, "3x3_threes.fits"),
+    )
+
+    assert (imaging.image.native == np.ones((3, 3))).all()
+    assert (imaging.psf.native == (1.0 / 9.0) * np.ones((3, 3))).all()
+    assert (imaging.noise_map.native == 3.0 * np.ones((3, 3))).all()
+
+    assert imaging.pixel_scales == (0.1, 0.1)
+    assert imaging.psf.mask.pixel_scales == (0.1, 0.1)
+    assert imaging.noise_map.mask.pixel_scales == (0.1, 0.1)
+
+    imaging = aa.Imaging.from_fits(
+        pixel_scales=0.1,
+        image_path=path.join(test_data_dir, "3x3_multiple_hdu.fits"),
+        image_hdu=0,
+        psf_path=path.join(test_data_dir, "3x3_multiple_hdu.fits"),
+        psf_hdu=1,
+        noise_map_path=path.join(test_data_dir, "3x3_multiple_hdu.fits"),
+        noise_map_hdu=2,
+    )
+
+    assert (imaging.image.native == np.ones((3, 3))).all()
+    assert (imaging.psf.native == (1.0 / 9.0) * np.ones((3, 3))).all()
+    assert (imaging.noise_map.native == 3.0 * np.ones((3, 3))).all()
+
+    assert imaging.pixel_scales == (0.1, 0.1)
+    assert imaging.psf.mask.pixel_scales == (0.1, 0.1)
+    assert imaging.noise_map.mask.pixel_scales == (0.1, 0.1)
+
+
+def test__output_to_fits():
+
+    imaging = aa.Imaging.from_fits(
+        pixel_scales=0.1,
+        image_path=path.join(test_data_dir, "3x3_ones.fits"),
+        psf_path=path.join(test_data_dir, "3x3_twos.fits"),
+        noise_map_path=path.join(test_data_dir, "3x3_threes.fits"),
+    )
+
+    output_data_dir = path.join(

```

Comment:
Make this a fixture. You can yield output_data_dir and delete it before and after yielding.

```python
@pytest.fixture(name='output_data_dir')
def make_output_data_dir(test_data_dir):
    output_data_dir = test_data_dir / 'array/output_test'
    shutil.rmtree(output_data_dir, ignore_errors=True)
    yield output_data_dir
    shutil.rmtree(output_data_dir, ignore_errors=True)

Reply:
Why do you want me to make a fixture for one test?

This always just feels confusing to me, and harder to trace what the test is doing.